Gems/Atom/RHI/Code/Source/RHI/Object.cpp

```diff
@@ -22,9 +29,14 @@ namespace AZ::RHI
         (void)name;
     }
 
-    const Name& Object::GetName() const
+    Name Object::GetName() const
     {
-        return m_name;
+        Name nameCopy;
+        {
+            AZStd::scoped_lock<decltype(m_nameMutex)> lock(m_nameMutex);
+            nameCopy = m_name;
```

Comment:
We don't have a read_write mutex class in the codebase ? It depends on the access pattern, but if Get is called a lot whereas Set only occurs from time to time, there should be a perf gain

Reply:
I was going to try that yesterday but I was tired, will give it a shot now and see if it makes a difference.

Reply:
No difference at all. Will try the `AZStd::spin_mutex` .

Reply:
What is the read_write mutex equivalent in o3de ? I didn't find anything when I looked for it

Reply:
AZStd::shared_mutex, And you can use unique_lock(mutex) or shared_lock(mutex).